### PR TITLE
Rubic aggregator

### PR DESCRIPTION
**Pull request by Doctor-Cyclone**
Created [Rubic](https://app.rubic.exchange/) volume adapter.

### CODE REVIEW

**Comment by dtmkeng** [comment]
your volume is count volume it go to other agg like 1inch ?

**Comment by Doctor-Cyclone** [comment]
- We have multi-proxy-rubic contracts. Explorer (https://etherscan.io/address/0x3335733c454805df6a77f825f266e136fb4a3333)
- Users come to our app. They make a swap. The swap goes through our contracts and only then through 1inch.
That is, we, as aggregators, use providers, but the volumes and users are ours, and they go through our contracts. We take into account the volumes of our contracts only.

**Comment by dtmkeng** [comment]
> * We have multi-proxy-rubic contracts. [Explorer](https://etherscan.io/address/0x3335733c454805df6a77f825f266e136fb4a3333)
> * Users come to our app. They make a swap. The swap goes through our contracts and only then through 1inch.
>   That is, we, as aggregators, use providers, but the volumes and users are ours, and they go through our contracts. We take into account the volumes of our contracts only.

how to exclude volume is go to other agg if list it double count with other dexs agg like 1inch

**Comment by Doctor-Cyclone** [comment]
Our volumes include volumes from 1inch and other aggregators if they were conducted through our proxy. No need to exclude anything

**Comment by dtmkeng** [comment]
could you count volume go to dexs not agge dexs like 1 inch

**Comment by Doctor-Cyclone** [comment]
Did not understand
Move the folder to Dex and not to aggregators?

**Comment by dtmkeng** [comment]
no i mean your api it count other agg volume ex 1 inch if list it double count. can you exclude other agg on your api

**Comment by Doctor-Cyclone** [comment]
Can you please tell me in more detail where the double count is? With us, all volumes pass through a proxy, without dividing providers into aggregator/non-aggregator

**Comment by dtmkeng** [comment]
> Can you please tell me in more detail where the double count is? With us, all volumes pass through a proxy, without dividing providers into aggregator/non-aggregator

Rubic is aggerator protocol and you tell me Rubic have some go too 1inch then this api `https://api.rubic.exchange/api/stats/defilama_onchain?dat` is include volume that go to 1inch right ? 
we can't list if you count volume that go to dexs agg intead dexs (uniswap, pancake...)

**Comment by asdener123** [comment]
> > Can you please tell me in more detail where the double count is? With us, all volumes pass through a proxy, without dividing providers into aggregator/non-aggregator
> 
> Rubic is aggerator protocol and you tell me Rubic have some go too 1inch then this api `https://api.rubic.exchange/api/stats/defilama_onchain?dat` is include volume that go to 1inch right ? we can't list if you count volume that go to dexs agg intead dexs (uniswap, pancake...)

Hey, could you please provide further details on why we should exclude DEX aggregators like 1inch from our volume calculations?
Our team strongly believes in including 1inch in our metrics, given its significant usage among Rubic users. Moreover, as a DEX aggregator ourselves, we're aggregating the best DEXs and Aggregators in our platform. 
As explained by @Doctor-Cyclone, we employ proxy contracts for each DEX or DEX aggregator, allowing us to earn commissions from swaps conducted through these platforms.
Excluding this data could result in inaccurate submissions to Defilama, affecting our volume, commissions, and ultimately misleading our users. Greatly appreciate any means to retain this data.
You can find the complete list of our integrated DEXs here: https://docs.rubic.finance/rubic/rubics-ecosystem

**Comment by Define101** [comment]
Hello @asdener123 and @Doctor-Cyclone 
To give you some context, we have 3 products relating to volume.
1- Normal volume (aka overview page) this is volume coming from swaps (dexes)
2- Derivatives volume (aka derivatives page) this is volume with leverage (basically, if there is leverage involved)
3- Dex aggregators volume (aka aggregators page), this is volume coming ONLY from dexes aggregators (witch themself use dexes, not other dexes aggregator)

Your product is a mix of dexes aggregator and a dexes aggregator aggregator (the product we have called llamaswap is a dexes aggregator aggregator) . For that reason, we don't track llamaswap volume since that volume is being tracked under the aggregators we use. With this in mind, there are 2 alternatives for rubi:
1)  We track ONLY volume coming from dexes (meaning your dexes aggregator product), and NOT the volume coming from dexes aggregators (like 1inch,paraswap e.t.c)2)  We don't track nothing

**Comment by Define101** [comment]
"Excluding this data could result in inaccurate submissions to Defilama, affecting our volume, commissions, and ultimately misleading our users. Greatly appreciate any means to retain this data." 

I understand this, but we can't break our own rules. We can maybe add a disclaimer, or something informing the user that we are only tracking your dex agg product and not the dex aggre aggre

**Comment by llamatester** [comment]
The rubic adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

**Comment by Doctor-Cyclone** [comment]
@Define101 
Hello, we are back to the process of adding Rubic to the DefiLlama. But when I try to run tests through the `yarn test aggregators rubic` command, I get the following error: `ERROR: Use Adapter v2 format for new adapters`

**Comment by Doctor-Cyclone** [comment]
We also excluded the following aggregators from our indicators: `openocean, xy_dex,  symbiosis, odos, lifi, rango, oneinch`

File: aggregators/rubic/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  return {
+    dailyVolume: responce?.daily_volume_in_usd || undefined,
+    totalVolume: responce?.total_volume_in_usd || undefined,
+    timestamp,
+  };
+};
+
+const adapter: SimpleAdapter = {
+  adapter: {
```
Comment (dtmkeng) [inline aggregators/rubic/index.ts:59]: add `version: 2` adove `adapter`

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: aggregators/rubic/index.ts (hunk 1 above)
Comment (Doctor-Cyclone) [reply aggregators/rubic/index.ts:59]: Done.

**Pull request merged**